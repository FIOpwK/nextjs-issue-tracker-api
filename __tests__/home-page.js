// import core dependencies
import React from "react";
import renderer from "react-test-renderer";
import Home from "../pages/home";

describe("Rendering content for the /home route", () => {
  beforeAll(() => {
    console.log("running...");
  });

  afterAll(() => [console.log("stopping...")]);

  test("page renders navigation links and  create new issue button", () => {
    const homePageComponent = renderer.create(<Home />).toJSON();
    expect(homePageComponent).toMatchSnapshot();
  });
});
