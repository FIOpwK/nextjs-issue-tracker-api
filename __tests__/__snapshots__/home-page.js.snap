// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering content for the /home route page renders navigation links and  create new issue button 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <h3>
       API Issue Tracker 
    </h3>
    <nav
      className="navbar sticky-top navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/home"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
           Home 
        </a>
        <a
          className="navbar-brand"
          href="/issues"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
           Issues 
        </a>
        <a
          className="navbar-brand"
          href="/#"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
           Tracker 
        </a>
        <a
          className="navbar-brand"
          href="/#"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
           News 
        </a>
        <a
          className="navbar-brand"
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
           About 
        </a>
      </div>
    </nav>
  </div>
  <div
    className="gap-3"
    style={
      Object {
        "margin": "5px",
      }
    }
  >
    <button
      className="btn btn-danger"
    >
       Open New Issue 
    </button>
  </div>
</div>
`;
